packages/core/src/main.ts

```diff
@@ -38,52 +38,46 @@ export async function createLibp2pInstance(
-      return a.sort((aa, ba) => compareAddressesLocalMode(aa.multiaddr, ba.multiaddr))
-    }
-    log('Preferring local addresses')
+): Promise<Libp2p> {
+  let libp2p: Libp2p
+  if (options.testing?.useMockedLibp2p) {
+    // Used for quick integration testing
+    libp2p = createLibp2pMock(peerId, {
+      network: options.testing.mockedNetwork,
+      dht: options.testing.mockedDHT
+    })
   } else {
-    // Overwrite address sorter with identity function since
-    // libp2p's own address sorter function is unable to handle
-    // p2p addresses, e.g. /p2p/<RELAY>/p2p-circuit/p2p/<DESTINATION>
-    addressSorter = (addr) => addr
-    log('Addresses are sorted by default')
-  }
+    let addressSorter: AddressSorter
+
+    if (options.testing?.preferLocalAddresses) {
+      addressSorter = (a: Address, b: Address) => compareAddressesLocalMode(a.multiaddr, b.multiaddr)
+      log('Preferring local addresses')
+    } else {
+      // Overwrite address sorter with identity function since
+      // libp2p's own address sorter function is unable to handle
+      // p2p addresses, e.g. /p2p/<RELAY>/p2p-circuit/p2p/<DESTINATION>
+      addressSorter = (_addr) => 0
+      log('Addresses are sorted by default')
+    }
 
-  // Store the peerstore on-disk under the main data path. Ensure store is
-  // opened before passing it to libp2p.
-  const datastorePath = path.join(options.dataPath, 'peerstore')
-  await mkdir(datastorePath, { recursive: true })
-  const datastore = new LevelDatastore(datastorePath, { createIfMissing: true })
-  await datastore.open()
-
-  log(`using peerstore at ${datastorePath}`)
-
-  const libp2p = await LibP2P.create({
-    peerId,
-    addresses: { listen: getAddrs(peerId, options).map((x) => x.toString()) },
-    // libp2p modules
-    modules: {
-      transport: [HoprConnect as any],
-      streamMuxer: [Mplex],
-      connEncryption: [NOISE as any],
-      dht: KadDHT
-    },
-    // Configure peerstore to be persisted using LevelDB, also requires config
-    // persistence to be set.
-    datastore,
-    peerStore: {
-      persistence: true
-    },
-    config: {
-      protocolPrefix: `hopr/${options.environment.id}`,
-      transport: {
-        HoprConnect: {
+    // Store the peerstore on-disk under the main data path. Ensure store is
+    // opened before passing it to libp2p.
+    const datastorePath = path.join(options.dataPath, 'peerstore')
+    await mkdir(datastorePath, { recursive: true })
+    const datastore = new LevelDatastore(datastorePath, { createIfMissing: true })
+    await datastore.open()
+
+    log(`using peerstore at ${datastorePath}`)
+
+    // Make libp2p aware of environments
+    const protocolPrefix = `/hopr/${options.environment.id}`
+
+    libp2p = await createLibp2p({
+      peerId,
+      addresses: { listen: getAddrs(peerId, options).map((x: Multiaddr) => x.toString()) },
+      transports: [
+        // @ts-ignore libp2p interface type clash
```

Comment:
to be solved, issues with  typing of `#private` methods and class inheritance in Typescript